.github/workflows/build.yml

```diff
@@ -0,0 +1,249 @@
+          asset_content_type: application/gzip
+
+      - name: deploy-to-pypi
+        shell: bash
+        run: |
+          python3 -m pip install twine==1.11.0
+
+          if [[ "${{ github.event.release.prerelease}}" == "false" ]]; then
+            # production deploy to prod pypi server
+
+            PYPI_NAME="pypi"
+            PYPI_URL="https://pypi.org"
+            PYPI_REPO="https://upload.pypi.org/legacy/"
+            PYPI_USERNAME="${{ secrets.PYPI_PROD_USERNAME }}"
+            PYPI_PASSWORD="${{ secrets.PYPI_PROD_PASSWORD }}"
+          else
+            # staging deploy to test pypi server
+
+            PYPI_NAME=testpypi
+            PYPI_URL="https://test.pypi.org"
+            PYPI_REPO=https://test.pypi.org/legacy/
+            PYPI_USERNAME="${{ secrets.PYPI_TEST_USERNAME }}"
+            PYPI_PASSWORD="${{ secrets.PYPI_TEST_PASSWORD }}"
+            PYPI_INDEX_URL="https://test.pypi.org/simple/"
+          fi
+
+          # create .pypirc file
+          echo "[distutils]" > ~/.pypirc
+          echo "index-servers=$PYPI_NAME" >> ~/.pypirc
+          echo >> ~/.pypirc
+          echo "[$PYPI_NAME]" >> ~/.pypirc
+          echo "repository: $PYPI_REPO" >> ~/.pypirc
+          echo "username:$PYPI_USERNAME" >> ~/.pypirc
+          echo "password:$PYPI_PASSWORD" >> ~/.pypirc
+
+          twine upload --repository $PYPI_NAME dist/*
+
+  # on each of our matrix platforms, download the newly uploaded package from pypi and confirm its version
+  check-deploy:
+    needs: package-deploy
+
+    strategy:
+      matrix:
+        os: [ubuntu-18.04, macos-10.15, windows-2019]
+        python: [3.6, 3.7, 3.8]
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python }}
+
+      - name: check-pypi
+        shell: bash
+        run: |
+          if [[ "${{ github.event.release.prerelease}}" == "false" ]]; then
+            PYPI_INDEX_URL="https://pypi.org/simple/"
+          else
+            PYPI_INDEX_URL="https://test.pypi.org/simple/"
+          fi
+
+          RELEASE_TAG="${{ github.event.release.tag_name }}"
+          if [[ $RELEASE_TAG =~ ^v?([[:digit:]\.]+)(-rc)? ]]; then
+            VERSION="${BASH_REMATCH[1]}"
+            if [[ "${{ github.event.release.prerelease}}" == "true" ]]; then
+              VERSION="$VERSION.$GITHUB_RUN_NUMBER"
+            fi
+          else
+            echo "Unrecognized release tag"
+            exit 1
+          fi
+
+          # it can take some time for the packages to become available in pypi after uploading
+          for i in 5 10 20 40; do
+            if pip3 install --index-url $PYPI_INDEX_URL --extra-index-url https://pypi.org/simple "synapseclient==$VERSION"; then
+              ACTUAL_VERSION=$(synapse --version)
+
+              if [ -n "$(echo "$ACTUAL_VERSION" | grep -oF "$VERSION")" ]; then
+                # found the expected version
+                exit 0
+              fi
```

Comment:
fail early if we've retrieved from pypi but version mismatch

Reply:
Done.